test/stream-monitor/flogging/infra_logging.py

```diff
@@ -0,0 +1,326 @@
+        finer-grained (more detailed) information than straight DEBUG.
+
+        This code matches anything for the form alpha-chars_n, which ends
+        up handling the following methods:
+        critical, fatal, error, warning, warn, info, debug, exception
+
+        Note that those base methods are resolved directly, and this method
+        is not called for those.
+        """
+        attr_match = self._log_call_matcher.match(key)
+        if attr_match is None:
+            m = "'{0}' object has no attribute '{1}'".format(self.__name__, key)
+            raise AttributeError(m)
+
+        base_name = attr_match.group("base").upper()
+        val_adj = int(attr_match.group("post_num"))
+        actual_value = _levelNames[base_name] - val_adj
+
+        def wrapper(msg, *args, **kw):
+            return self._log(actual_value, msg, args, kw)
+
+        # We return a wrapper function, since we are being asked to resolve
+        # the method, not call it. The above wrapper allows us to return
+        # something callable that still allows us to inject the calculated
+        # level-value.
+        return wrapper
+
+
+class _LoggerSetup(object):
+    def __init__(self):
+        self.__prelog_data = []
+        self.__do_dirs()
+        self.__do_level_names()
+        self.__do_config()
+        lg = getInfraRunLogger()
+        for level, fmat, args, kwargs in self.__prelog_data:
+            lg.log(level, fmat, *args, **kwargs)
+
+    def __prelog(self, level, fmat, *args, **kwargs):
+        self.__prelog_data.append( (level, fmat, args, kwargs) )
+
+    def __makedirs_dash_p(self, *args, **kwargs):
+        try:
+            os.makedirs(*args, **kwargs)
+        except OSError, e:
+            # be happy if someone already created the path
+            if e.errno != errno.EEXIST:
+                raise
+
+    def __do_dirs(self):
+        our_dir = os.path.dirname(os.path.abspath(__file__))
+        test_dir = os.path.dirname(os.path.dirname(our_dir))
+        lg_base_dir = os.path.join(test_dir, 'log_output')
+        self.__makedirs_dash_p(lg_base_dir)
+        dirs_in_log = os.listdir(lg_base_dir)
+        matcher = re.compile(r'''^                            # SOL
+                             run_                             # 'run_'
+                             (?P<run_date>\d\d\d\d-\d\d-\d\d  # 2016-11-18
+                             _                                # '_'
+                             \d\d:\d\d:\d\d)                  # 23:11:00
+                             \.d$''',                         # '.d' EOL
+                             re.VERBOSE)
+        matches = {}
+        for file_name in dirs_in_log:
+            m = matcher.match(file_name)
+            if m is not None:
+                key = m.group("run_date")
+                matches[key] = file_name
+
+        keys = sorted(matches.keys(), reverse=True)
+        trim_these = keys[3:]
+        for trim_inx in trim_these:
```

Comment:
Sure, but it's the only change you get to request :). (I was going to make it an env variable, but decided there will be other things we want to configure on this and it made sense to not keep adding stuff just now...)